src/Platform/Microsoft.Testing.Platform/CommandLine/ICommandLineOptions.cs

```diff
@@ -23,3 +23,24 @@ public interface ICommandLineOptions
     /// <returns>True if the argument list is found; otherwise, false.</returns>
     bool TryGetOptionArgumentList(string optionName, [NotNullWhen(true)] out string[]? arguments);
 }
+
+internal static class CommandLineOptionsExtensions
+{
+    public static bool TryGetOptionArgument(this ICommandLineOptions options, string optionName, [NotNullWhen(true)] out string? argument)
+    {
+        if (!options.TryGetOptionArgumentList(optionName, out string[]? arguments) ||
+            arguments.Length == 0)
+        {
+            argument = null;
+            return false;
+        }
```

Comment:
I would `return false` only if `TryGetOptionArgumentList` returned false. Otherwise, throw if `arguments.Length != 1`, and I would probably throw Unreachable, since validation should have happened early and by the point we are calling this, we should be expecting us to be in a valid state.